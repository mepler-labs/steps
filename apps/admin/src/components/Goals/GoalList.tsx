import React from 'react';
import Goal from './Goal';
import styled from 'styled-components';

interface Props {
  className?: string;
  goals: string[];
}

const GoalList: React.SFC<Props> = ({ goals }) => (
  <div>
    {goals.map((goal, i) => (
      <StyledGoal text={goal} key={i} onEdit={() => {}} />
    ))}
  </div>
);

export const StyledGoal = styled(Goal)`
  margin-bottom: 2rem;
`;

export default GoalList;
