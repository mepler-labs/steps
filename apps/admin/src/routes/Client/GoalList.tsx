import Goal from 'components/Goals/Goal';
import { GoalList } from 'components/Goals/GoalList';
import React from 'react';
import { connect, History } from 'react-redux';

interface Props {
  goals: string[];
}

export class Goals extends React.Component<Props> {
  render() {
    const { goals } = this.props;
    return (
      <GoalList goals={goals}>
        {childProps => <Goal {...childProps} />}
      </GoalList>
    );
  }
}
const mapStateToProps = state => ({
  goals: state.auth.user.goals,
});

export default connect(mapStateToProps)(Goals);
