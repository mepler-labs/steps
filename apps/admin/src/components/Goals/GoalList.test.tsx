import { shallow } from 'enzyme';
import 'jest';
import * as React from 'react';
import GoalList, { StyledGoal } from './GoalList';

describe('GoalList.tsx', () => {
  it('renders correctly', () => {
    const goals = ['goal 1', 'goal 2', 'goal 3'];
    const wrapper = shallow(<GoalList goals={goals} />);

    expect(wrapper).toMatchSnapshot();
  });
});
