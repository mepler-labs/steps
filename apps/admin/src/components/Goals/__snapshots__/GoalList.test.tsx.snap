// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalList.tsx renders correctly 1`] = `
<div>
  <Styled(Goal)
    key="0"
    onEdit={[Function]}
    text="goal 1"
  />
  <Styled(Goal)
    key="1"
    onEdit={[Function]}
    text="goal 2"
  />
  <Styled(Goal)
    key="2"
    onEdit={[Function]}
    text="goal 3"
  />
</div>
`;
